rclcpp/include/rclcpp/generic/typesupport_helpers.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__GENERIC__TYPESUPPORT_HELPERS_HPP_
```

Comment:
This might be a bit more work, but I really would see this functionality being placed in `rcpputils`. The type support loading has more applications than just the generic publishing and could technically also be used independently of `rclcpp`.

Reply:
not a show-stopper for this PR though.

Reply:
I don't think this is appropriate for `rcpputils`, as it's not just a "CPP" thing, it's clearly a ROS thing, and I think `rcutils` and `rcpputils` are not meant to be so ROS centric.

Reply:
`rcl` perhaps for part of it, but the C++ part of it should be in `rclcpp` I think. Anyways I made similar comments in my review too.